src/util/createServerRootMixin.js

```diff
@@ -68,32 +100,7 @@ function augmentInstantSearch(instantSearchOptions, searchClient, indexName) {
 
     return Promise.resolve()
       .then(() => {
-        const options = {
-          serverPrefetch: undefined,
-          fetch: undefined,
-          _base: undefined,
-          name: 'ais-ssr-root-component',
-          // copy over global Vue APIs
-          router: componentInstance.$router,
-          store: componentInstance.$store,
-        };
-
-        const Extended = componentInstance.$vnode
-          ? componentInstance.$vnode.componentOptions.Ctor.extend(options)
-          : Vue.component(
-              Object.assign({}, componentInstance.$options, options)
-            );
-
-        app = new Extended({
-          propsData: componentInstance.$options.propsData,
-        });
-
-        // https://stackoverflow.com/a/48195006/3185307
-        app.$slots = componentInstance.$slots;
-
-        app.$root = componentInstance.$root;
-
-        app.$options.serverPrefetch = [];
+        app = cloneComponent(componentInstance);
```

Comment:
what do you think of cloneComponent vs (disclaimer: temp name) augmentApp, where

augmentApp = (app, componentInstance) => app

so that user can benefit from our default implementation or just ignore and create a fresh app instance based on componentInstance if they want.

Reply:
app is the name of the component though here, not an "app", just `app = new Vue` is a common naming. Cloning the component isn't optional, it's more that they might want to copy more items than what we did in the default one (like i18n in the linked issue)

Reply:
makes sense!